tests/test_model_spatial.py

```diff
@@ -20,28 +20,28 @@ def training_data():
 
 
 @pytest.fixture()
-def env_image():
+def X_image():
     return get_covariate_image()
 
 
-def get_check_img(prefix: str) -> ee.Image:
+def get_check_image(prefix: str) -> ee.Image:
     return ee.Image(f"users/gregorma/gee-knn/test-check/{prefix}_neighbors_600")
 
 
-def run_method(kls, options, training_data, env_image, check_img):
+def run_method(kls, options, training_data, X_image, check_image):
     """Run predict on the given estimator, difference it against a
     reference images, and return the frequency of zero differences"""
     model = kls(**options).train(**training_data)
-    nn = model.predict(env_image=env_image, mode="CLASSIFICATION").retile(32)
-    diff_nn = nn.subtract(check_img).abs()
-    frequency = diff_nn.reduceRegion(
+    nn_image = model.predict(X_image=X_image, mode="CLASSIFICATION").retile(32)
```

Comment:
Yeah, I'm pretty sure I'm not supposed to be advertising this one (or maybe even using it). This was something they put in the API a while back, but never really documented it (nor took it away!). AFAIK, `tileScale` can only be used with reducing or sampling(?), but I think it's the same concept. I believe the way Noel explained it to me was that `retile` just makes the map tiles smaller so that you're passing less data and, therefore, less likely to run into memory issues. In this case, if I don't use it, tests don't pass and I get `User memory limit exceeded` errors. If we were doing exports, I believe this wouldn't be an issue (different memory limits, correct?).

But I think the inclusion of `tileScale` in the `reduceRegion` call is now not needed because we have this `.retile` call in there.

Reply:
Very cool, thanks for the inside scoop! I won't spread the word, but I am curious to try it out. I wonder why it didn't make it into the public methods, since it seems like a more flexible approach than `tileScale`.